python/nano/src/bigdl/nano/deps/ipex/ipex_inference_model.py

```diff
@@ -124,33 +129,68 @@ def __init__(self, model: torch.nn.Module, input_sample=None, use_ipex=False, dt
                 with torch.no_grad():
                     with torch.cpu.amp.autocast():
                         if self.jit_method == 'trace':
-                            self.model = torch.jit.trace(self.model, input_sample,
-                                                         check_trace=False,
-                                                         strict=jit_strict)
+                            if compare_version("torch", operator.ge, "2.0"):
+                                self.model = torch.jit.trace(
+                                    self.model,
+                                    example_inputs=input_sample,
+                                    check_trace=False,
+                                    strict=jit_strict,
+                                    example_kwarg_inputs=example_kwarg_inputs)
+                            else:
+                                self.model = torch.jit.trace(
+                                    self.model, input_sample,
+                                    check_trace=False,
+                                    strict=jit_strict)
                         elif self.jit_method == 'script':
                             self.model = torch.jit.script(self.model)
                         else:
                             try:
-                                self.model = torch.jit.trace(self.model, input_sample,
-                                                             check_trace=False,
-                                                             strict=jit_strict)
+                                if compare_version("torch", operator.ge, "2.0"):
+                                    self.model = torch.jit.trace(
+                                        self.model,
+                                        example_inputs=input_sample,
+                                        check_trace=False,
+                                        strict=jit_strict,
+                                        example_kwarg_inputs=example_kwarg_inputs)
+                                else:
+                                    self.model = torch.jit.trace(self.model, input_sample,
+                                                                 check_trace=False,
+                                                                 strict=jit_strict)
                             except Exception:
                                 self.model = torch.jit.script(self.model)
                         if self.use_ipex:
                             self.model = torch.jit.freeze(self.model)
             else:
```

Comment:
There is too much redundant code here.
For example, this `if` `else`  can be removed, you only need to remove line 128 and this `else` branch, then change line 130 to
```python
with torch.cpu.amp.autocast(enabled=dtype == torch.bfloat16):
```

Reply:
Good suggestion, thanks! Have revised to this form.